pkg/sql/row/errors.go

```diff
@@ -207,11 +212,21 @@ func decodeRowInfo(
 	// Use the Fetcher to decode the single kv pair above by passing in
 	// this singleKVFetcher implementation, which doesn't actually hit KV.
 	if err := rf.StartScanFrom(ctx, &f); err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	datums, _, _, err := rf.NextRowDecoded(ctx)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
+	}
+
+	names := make([]string, len(cols))
+	values := make([]string, len(cols))
+	for i := range cols {
+		names[i] = cols[i].Name
+		if datums[i] == tree.DNull {
```

Comment:
why not write `NULL` into the string here? I feel like we depend on this behavior in some error messages

Reply:
I don't think it comes up for existing users, just the new allColumns case. Tests pass?